model/src/main/proto/exploration.proto

```diff
@@ -0,0 +1,145 @@
+syntax = "proto3";
+
+import "subtitled_html.proto";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Data for a single exploration.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeExploration.kt
+message Exploration {
+  repeated State states = 1;
+  repeated ParamChange param_changes = 2;
+  repeated ParamSpec param_specs = 3;
+  string initStateName = 4;
+  string objective = 5;
+  bool correctness_feedback_enabled = 6;
+  string title = 7;
+  string language_code = 8;
+}
+
+// Data for a param change.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamChange.kt
+message ParamChange {
+  string generator_id = 1;
+  string name = 2;
+  CustomizationArgs customization_args = 3;
+}
+
+// Data for a param spec.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamSpec.kt
+message ParamSpec {
+  string obj_type = 1;
```

Comment:
Can we document which types of objects can be expected here? Could this be represented via an enum, instead?

Reply:
Done